binderhub/builder.py

```diff
@@ -255,6 +255,24 @@ def escape(s):
             await self.fail("Could not resolve ref for %s. Double check your URL." % key)
             return
 
+        self.ref_url = await provider.get_resolved_ref_url()
+        resolved_spec = await provider.get_resolved_spec()
+
+        self.binder_url = '{proto}://{host}{base_url}v2/{provider}/{spec}'.format(
```

Comment:
We should check if we need to/should use https://github.com/jupyterhub/binderhub/blob/2a7145f40dfc9d9face452d3f3f4502eeb4392a2/binderhub/app.py#L149-L153 instead of the request host. `badge_base_url` is used by hubs in the federation to always generate links with `https://mybinder.org/...` instead of the individual cluster's host name.

Reply:
The current "Copy Binder Link" button already seems to only generate links that start with `https://mybinder.org/...`. I think this is because we try and do some clever stuff in the JS. Maybe we can simplify that if we generate the "right" URL here already?